src/python/numpyConversion.hpp

```diff
@@ -0,0 +1,119 @@
+/*
+ *~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ * Copyright (c) 2019, Lawrence Livermore National Security, LLC.
+ *
+ * Produced at the Lawrence Livermore National Laboratory
+ *
+ * LLNL-CODE-746361
+ *
+ * All rights reserved. See COPYRIGHT for details.
+ *
+ * This file is part of the GEOSX Simulation Framework.
+ *
+ * GEOSX is a free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License (as published by the
+ * Free Software Foundation) version 2.1 dated February 1999.
+ *~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ */
+
+/**
+ * @file numpyConversion.hpp
+ */
+
+#pragma once
+#define PY_SSIZE_T_CLEAN
+#define NPY_NO_DEPRECATED_API NPY_1_15_API_VERSION
+#define NO_IMPORT_ARRAY
+#define PY_ARRAY_UNIQUE_SYMBOL LvArray_ARRAY_API
+
+#include <Python.h>
+#include <numpy/arrayobject.h>
+
+#include "../StringUtilities.hpp"
+#include "../IntegerConversion.hpp"
+
+#include <typeindex>
+#include <type_traits>
+
+namespace LvArray
+{
+
+namespace python
+{
+
+namespace internal
+{
+
+/**
+ * @brief Get the Numpy type-flag corresponding to a `std::type_info` object.
+ * @param tid the type_info object to use
+ */
+int getFlags( const std::type_info& tid ){
+    if ( std::type_index( tid ) == std::type_index( typeid( char ) ) )
```

Comment:
Can you add unsigned variations on all of these if numpy supports it? Be sure to have both `signed` and `unsigned char` since the signedness of `char` is implementation defined.